plugins/woocommerce/src/Admin/DataSourcePoller.php

```diff
@@ -133,7 +133,10 @@ public function read_specs_from_data_sources() {
 			$this->merge_specs( $specs_from_data_source, $specs, $url );
 		}
 
-		$specs_group            = get_transient( $this->args['transient_name'] ) ?? array();
+		$specs_group = get_transient( $this->args['transient_name'] );
+		if ( ! is_array( $specs_group ) ) {
```

Comment:
Strictly speaking, a direct comparison to `false` should be used here if we want to replicate the intended behavior of "empty array if the transient doesn't exist" (as it could exist but have a non-array value). However this would imply throwing (or at least logging) an error, which is probably overkill since that's never supposed to happen.

Reply:
@konamiman—you're suggesting a direct comparison to false as in the following example?

```suggestion
		if ( is_array( $specs_group ) === false ) {
```